tensorflow/c/eager/c_api.cc

```diff
@@ -111,7 +111,7 @@ tensorflow::Status GetAllRemoteDevices(
 tensorflow::Status CreateRemoteContexts(
     const std::vector<string>& remote_workers, int64 rendezvous_id,
     const tensorflow::ServerDef& server_def,
-    tensorflow::eager::EagerClientCache* remote_eager_workers, bool async,
+    tensorflow::eager::EagerClientCache* remote_eager_workers, bool is_async,
```

Comment:
If `async` is a reserved word in Python 3.7, I don't quite follow why all these C files have to change :)
(`c_api.cc`, `c_api.h`, `context.cc`, `context.h` etc.)

Let's revert the changes to the C files?

Reply:
``c_api.cc`` and ``c_api.h`` have to change because ``pywrap_tensorflow_internal.py`` is generated from them, however I found that change should be rather minimal and touch only ``TFE_ContextOptionsSetAsync()`` and ``TFE_ContextSetAsyncForThread()``

Reply:
Shouldn't this be handled in `pywrap_tfe.i` instead of changing the C code?

Reply:
You are right, this may be handled in ``pywrap_tfe.i`` but from my small investigation I could solve this issue only with some sort of hack - define macro to redefine variable name, i.e. put
```
#define async is_async
```
into ``pywrap_tfe.i``. ``%rename`` function didn't help because it works only with declarations but not with arguments.

Reply:
Would writing a wrapper function in the SWIG file work?
(Not for the same reason, but we hide `TF_SessionRun` from Python and instead use `TF_SessionRun_wrapper` in https://github.com/tensorflow/tensorflow/blob/b8a9d163d9cbb4b581c044d9c4b1b256c801a9c4/tensorflow/python/client/tf_session.i#L482 )

Reply:
I thought of wrapping and I think it should work and it is more direct way than defining macro. But it will require watching for changes in tensorflow c api - watching for updated signatures of wrapped functions and for appearing ``async`` parameter in functions.

Reply:
You are right, there are only 2 functions and one should watch for new signatures in case of updated C code anyway.